src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -241,6 +251,17 @@ class Crystal::Doc::Generator
     end
   end
 
+  # adds extra line break to TODO lines
+  def isolate_flag_lines(string)
+    string.split(/\n/).map do |line|
```

Comment:
However my first suggestion with the each_line and join with block are still valid with this workaround.